backend/volume/src/chat/chat.gateway.ts

```diff
@@ -23,101 +26,194 @@ export class ChatGateway implements OnGatewayInit, OnGatewayConnection, OnGatewa
   constructor(
 		private msgService: MsgService,
 		private roomService: RoomService,
+    private userService: PrismaUserService,
+    private gate: GateService,
+    private chatService: ChatService,
+    private jwtService: JwtService,
   ){}
   private server: Server;
+  // private gateService: GateService;
 
   @WebSocketServer() all_clients: Server; //all clients
   
   @SubscribeMessage('sendMsg')
-  handleMessage(client: Socket, packet: any) {
-    const id = packet.id;
-    // const user = packet.username;
-    const text = packet.msg;
-    this.all_clients.emit('receiveNewMsg', packet);
-	
-    const dto: MsgDto = {id: -1, roomId: 1, body: text, authorId: id, invite: false};
-    this.msgService.handleIncomingMsg(dto);  // handles db placement of the new msg based on sender id
+  async handleMessage(client: Socket, packet: MsgDto) {
+    // input protection
+    if (packet.roomId === undefined || packet.body === undefined)
+      return ;
+    if (packet.invite === undefined)
+      packet.invite = false;
+
+    const userId = await this.gate.getUserBySocket(client);
+
+    packet.authorId = userId;
+
+    const msgWithAuthor = await this.msgService.handleIncomingMsg(packet);  // handles db placement of the new msg based on sender id
+
+    // add functionality to send the message to the users in that chat who are online
+    // temporarely send to everyone connected !!! not for end production
+    this.spreadMessage(msgWithAuthor);
   }
 
   afterInit(server: Server) {
     this.server = server;
-    console.log('Gateway initialised.');
   }
 
-  handleConnection(client: Socket) {
-    // client.broadcast.emit('msgToClient', this.formatMessage('Rubot','A new user joined the chat.')); //all OTHER clients
-    // client.emit('msgToClient', this.formatMessage('Rubot','Welcome to the chat!')); //this client
-    // console.log(`Client ${client.id} connected`);
+  async handleConnection(client: Socket) {
+    if (client.handshake.auth.token === '')
+      return;
+    const user = await this.jwtService.verify(
+      client.handshake.auth.token, { secret: process.env.JWT_SECRET }
+    );
+    // if error maybe we need to recover or exit or something!!!
 
-    client.emit('loadAllChats'); // data, all chats roomDto[]
+    // waarom is id 'sub' ???
```

Comment:
`.verify()` returns a payload object, not a user object. This is also why `sub` contains the user id. As it is the Subject of the payload.